controller/engine_image_controller.go

```diff
@@ -254,10 +254,11 @@ func (ic *EngineImageController) syncEngineImage(key string) (err error) {
 		return errors.Wrapf(err, "cannot get daemonset for engine image %v", engineImage.Name)
 	}
 	if ds == nil {
-		tolerations, err := ic.ds.GetSettingTaintToleration()
+		setting, err := ic.ds.GetSetting(types.SettingNameTaintToleration)
```

Comment:
Is there any other place using the original function?

Reply:
Yes, there are a few places. I modified them all except for the this place https://github.com/longhorn/longhorn-manager/blob/8af120dd7c109d81d86118e482f88a26ddf7650f/controller/volume_controller.go#L1982. I think that it is ok to keep it the same here.

Reply:
Hmm, we don't need this change in cronjob?

Reply:
The current design for cron job is that if any field in cron job' spec is different than the desired spec, we update the cron job to the desire state,[link](https://github.com/longhorn/longhorn-manager/blob/8af120dd7c109d81d86118e482f88a26ddf7650f/controller/volume_controller.go#L2083). 
This logic already cover the modification to tolerations list. 

I know that this would be a problem if K8s automatically fill in some default values that is not specify by the desired spec. In this case, Longhorn will keep creating/deleting cron job forever. So far we don't hit this issue yet. If we hit it, it should be an other issue?

Reply:
It seems safer if we use the same annotation method? Of course, then we need to change the `deepEquals` a bit.

Reply:
@yasker 
Actually, the current logic for comparison is not correct. This check always fails:
https://github.com/longhorn/longhorn-manager/blob/8af120dd7c109d81d86118e482f88a26ddf7650f/controller/volume_controller.go#L2083
And  Longhorn constantly updates cronjobs every reconciliation cycle. 

I have re-write the comparison logic using annotation.